code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -156,8 +156,14 @@
 	if(!I || !user)
 		return 0
 
-	var/obj/item/bodypart/affecting = get_bodypart(ran_zone(user.zone_selected)) //what we're actually ending up trying to hit.
+	var/obj/item/bodypart/affecting
+	if(user == src)
+		to_chat(user, "stab!")
+		affecting = get_bodypart(user.zone_selected) //stabbing yourself always hits the right target

```

Comment:
you can't directly use a zone.
```ran_zone()``` uses ```check_zone()``` to validate it, you're now no longer validating the zone.